compiler/ballerina-parser/src/main/java/io/ballerina/compiler/syntax/tree/NodeParser.java

```diff
@@ -17,18 +17,35 @@
  */
 package io.ballerina.compiler.syntax.tree;
```

Comment:
We can move `NodeParser` to `io.ballerina.compiler.syntax` pkg also(together with SyntaxInfo class). since this is not actually a part of the syntax tree?

Reply:
btw this seem to require making `NodeList(NonTerminalNode node)` constructor public, since we are using it in `NodeParser` to return a `NodeList<StatementNode>`. Aslo `SeparatedNodeList(NonTerminalNode node)` for the consistency.